### PR TITLE
Add propagators configuration to the Instrumentation kind

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

**Comment by jpkrohling** [comment]
This needs a rebase.

File: apis/instrumentation/v1alpha1/instrumentation_types.go [hunk 1]
```diff
@@ -28,6 +28,11 @@ type InstrumentationSpec struct {
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	ResourceAttributes map[string]string `json:"resourceAttributes,omitempty"`
 
+	// Propagators defines inter-process context propagation configuration.
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	Propagators []string `json:"propagators,omitempty"`
```
Comment (jpkrohling) [inline apis/instrumentation/v1alpha1/instrumentation_types.go:34]: Isn't there an annotation that generates some validation of the possible values? I know we have that for enums, but I wonder if there's something for a list of values?

File: pkg/instrumentation/sdk.go [hunk 2]
```diff
@@ -82,9 +83,11 @@ func injectCommonSDKConfig(otelinst v1alpha1.Instrumentation, ns corev1.Namespac
 	}
 	idx = getIndexOfEnv(container.Env, envOTELPropagators)
 	if idx == -1 && len(otelinst.Spec.Propagators) > 0 {
+		specialScopes := *(*[]string)((unsafe.Pointer(&otelinst.Spec.Propagators)))
```
Comment (jpkrohling) [inline pkg/instrumentation/sdk.go]: This now has the potential to break the operator: could you please add some negative tests? Perhaps isolate this in its own func to facilitate testing.

Comment (pavolloffay) [reply pkg/instrumentation/sdk.go]: > This now has the potential to break the operator

Could you please explain why it has the potential to break the operator?

Comment (jpkrohling) [reply pkg/instrumentation/sdk.go]: I need to play around, but you have a few conversions and pointer operations happening there. If you are confident that they are safe with all possible types of inputs, then I'm fine with it.

Comment (pavolloffay) [reply pkg/instrumentation/sdk.go]: I am fine with this.

You can do the exercise and get back here if this is risky.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**